test/Microsoft.ML.Tests/Transformers/PcaTests.cs

```diff
@@ -14,47 +13,85 @@ namespace Microsoft.ML.Tests.Transformers
 {
     public sealed class PcaTests : TestDataPipeBase
     {
+        private readonly ConsoleEnvironment _env;
+        private readonly string _dataSource;
+        private readonly TextSaver _saver;
+
         public PcaTests(ITestOutputHelper helper)
             : base(helper)
         {
+            _env = new ConsoleEnvironment(seed: 1, conc: 1);
```

Comment:
>_env = new ConsoleEnvironment(seed: 1, conc: 1); [](start = 12, length = 48)

How necessary is to have this environment?
I believe you specify seed in arguments, so you don't have to specify seed here.
Does concurrency affect results of PCA?

Reply:
concurrency didn't have an effect, on my Windows machine at least, so I removed it.

The seed value is needed and it does change the output. PCA seed doesn't seem to work and I opened an issue #1337 for it.
[https://github.com/dotnet/machinelearning/issues/1337](https://github.com/dotnet/machinelearning/issues/1337)

---
In reply to: [227047586](https://github.com/dotnet/machinelearning/pull/1333#discussion_r227047586) [](ancestors = 227047586)